doc/http-client.md

```diff
@@ -40,93 +41,491 @@ There are a few important caveats to mention about this library:
 
 ```haskell
 #!/usr/bin/env stack
--- stack --install-ghc --resolver lts-5.10 runghc --package http-client-tls
-import Network.HTTP.Client
-import Network.HTTP.Client.TLS   (tlsManagerSettings)
-import Network.HTTP.Types.Status (statusCode)
+-- stack --install-ghc --resolver lts-5.10 runghc --package http-conduit
+{-# LANGUAGE OverloadedStrings #-}
+import qualified Data.ByteString.Lazy.Char8 as L8
+import           Network.HTTP.Simple
 
 main :: IO ()
 main = do
-    manager <- newManager tlsManagerSettings
+    response <- httpLBS "http://httpbin.org/get"
 
-    request <- parseUrl "http://httpbin.org/get"
-    response <- httpLbs request manager
+    putStrLn $ "The status code was: " ++
+               show (getResponseStatusCode response)
+    print $ getResponseHeader "Content-Type" response
+    L8.putStrLn $ getResponseBody response
+```
+
+`httpLBS` makes a request to the given URL and captures the response body as a
+lazy `ByteString`. Note that, even though this is a lazy `ByteString`, it is
+read fully into memory when making the request. It only returns a lazy
+`ByteString` for better memory usage. (See streaming below for more information.)
+
+Once we have our response value, we can use getter functions to look at various
+details (status code, headers, and the body).
+
+## Receiving JSON
+
+We can also leverage aeson to receive a JSON message.
+
+```haskell
+#!/usr/bin/env stack
+-- stack --install-ghc --resolver lts-5.10 runghc --package http-conduit
+{-# LANGUAGE OverloadedStrings #-}
+import           Data.Aeson            (Value)
+import qualified Data.ByteString.Char8 as S8
+import qualified Data.Yaml             as Yaml
+import           Network.HTTP.Simple
+
+main :: IO ()
+main = do
+    response <- httpJSON "http://httpbin.org/get"
 
     putStrLn $ "The status code was: " ++
-               show (statusCode $ responseStatus response)
-    print $ responseBody response
+               show (getResponseStatusCode response)
+    print $ getResponseHeader "Content-Type" response
+    S8.putStrLn $ Yaml.encode (getResponseBody response :: Value)
```

Comment:
I think this is a good example of where a nice display typeclass is missing.

https://github.com/commercialhaskell/jump/issues/11

I use to code in ruby (using pry as repl), and default output was great for most types / object, with colors when possible, nice indentation, etc.

![image](https://cloud.githubusercontent.com/assets/2150990/14557780/36042624-02ff-11e6-85dc-7fe386355a1d.png)